dataset_loaders/videos/davis2017.py

```diff
@@ -0,0 +1,279 @@
+    @property
+    def filenames(self):
+        if self._filenames is None:
+            self._filenames = {}
+            # Get file names for this set
+            for vid_dir in self.prefix_list:
+                self._filenames[vid_dir] = []
+                for root, dirs, files in os.walk(os.path.join(
+                        self.image_path, vid_dir)):
+                    for name in files:
+                        self._filenames[vid_dir].append(os.path.join(
+                            os.path.split(root)[1], name[:-4]))
+                self._filenames[vid_dir].sort(key=natural_keys)
+        return self._filenames
+
+    def __init__(self,
+                 which_set='train',
+                 multiobject=True,
+                 dataset_version='2017',
+                 threshold_masks=False,
+                 split=.75,
+                 *args, **kwargs):
+
+        """ The Davis 2017 dataset
+        The following loader allows to load both the 2016 and 2017
+        version of the Davis dataset.
+
+        Parameters
+        ----------
+        which_set: string
+            A string in ['train', 'val', 'test-dev'], corresponding to
+            the training, validation and test sets specifically.
+        multiobject: boolean
+            In the case of the last dataset version (2017), this
+            parameter allows to decide when to return the labels in
+            multi-object notation (more then one object instance) or in the
+            binary one (only two ids 0-1).
+        dataset_version: string
+            A string in ['2016', '2017']. It allows to select the
+            version of Davis dataset to be loaded. Depending on this
+            parameter the files will be loaded from the most recent dataset
+            folder (2017) or from the previous one (2016).
+
+        """
+
+        if which_set not in {"train", "val", "test-dev"}:
+            raise ValueError("Unknown set {}".format(which_set))
+        self.which_set = which_set
+        self.threshold_masks = threshold_masks
+        self.multiobject = multiobject
+        if dataset_version not in ['2016', '2017']:
+            raise RuntimeError('Unknown dataset version')
+        self._image_sets_path = os.path.join(self.path, 'ImageSets',
+                                             dataset_version)
+        self.image_path = os.path.join(self.path,
+                                       'JPEGImages', '480p')
+        self.mask_path = os.path.join(self.path,
+                                      'Annotations', '480p')
+        self.rgb_values_shared_path = os.path.join(
+            self.shared_path, 'rgb_values_' + dataset_version + '.npy')
+        self.rgb_values_local_path = os.path.join(
+            self.path, 'rgb_values_' + dataset_version + '.npy')
+        if not os.path.exists(self.rgb_values_shared_path):
+            self.save_rgbs()
+
+        # Prepare data paths
+        self.split = 1. if self.which_set == 'test-dev' else split
```

Comment:
self.split is never used. I believe there is an official list of validation files. Is that so? If so, remove the split argument. Otherwise, make sure to use self.split where you load the filenames.